### ISSUE TITLE
Destructuring custom types with labeled and unlabeled field crashes the compiler with JS target

### ISSUE THREAD

**Issue by jiegillet** [gleam-lang/gleam#2007]
The following program

```gleam
type Foo {
  Foo(labeled: Int, Int)
}

fn bar(foo: Foo) -> Int {
  let Foo(labeled, unlabeled) = foo 
  labeled
}
```

Makes the compiler crash when using ` --target=javascript` with the following message:

```
➜  hello git:(main) ✗ gleam run --target=javascript
  Compiling hello
error: Fatal compiler bug!

This is a bug in the Gleam compiler, sorry!

Please report this crash to https://github.com/gleam-lang/gleam/issues/new
with this information and the code that produces the crash.

	compiler-core/src/javascript/pattern.rs:403
	argument present in field map
```

I suspect it happens during the destructuring the custom types with some labeled fields and some unlabeled fields.
The compiler crashes before writing the `.mjs` file, but if I replace the function body with a `todo`, we get

```js
import { CustomType as $CustomType, throwError } from "./gleam.mjs";

class Foo extends $CustomType {
  constructor(labeled, x1) {
    super();
    this.labeled = labeled;
    this[1] = x1; 
  }
}

function bar(foo) {
  throwError("todo", "hello", 5, "bar", "This has not yet been implemented", {});
}
```

Gleam version:
```
➜  hello git:(main) ✗ gleam --version                         
gleam 0.26.0
```

**Comment by JohnBjrk** [comment]
I think this was solved for (unreleased) #1982

**Comment by lpil** [comment]
I think so! Thank you
